qt_gui_cpp/src/qt_gui_cpp_sip/CMakeLists.txt

```diff
@@ -31,6 +31,39 @@ set(qt_gui_cpp_sip_INCLUDE_DIRS ${qt_gui_cpp_INCLUDE_DIRS} "${CMAKE_CURRENT_SOUR
 set(qt_gui_cpp_sip_LIBRARY_DIRS ${qt_gui_cpp_LIBRARY_DIRS} lib)
 set(qt_gui_cpp_sip_LDFLAGS_OTHER ${qt_gui_cpp_LDFLAGS_OTHER} -Wl,-rpath,\\"lib\\")
 
+find_library(class_loader_LIBRARY class_loader)
+find_library(console_bridge_LIBRARY console_bridge)
+find_library(TinyXML2_LIBRARY tinyxml2)
+
+get_filename_component(qt_gui_cpp_src_DIR ${CMAKE_CURRENT_BINARY_DIR} DIRECTORY)
+set(qt_gui_cpp_LIBRARY ${qt_gui_cpp_src_DIR}/qt_gui_cpp/Release/qt_gui_cpp${CMAKE_STATIC_LIBRARY_SUFFIX})
+
+set(_qt_gui_cpp_sip_LIBRARIES
+  ${PYTHON_LIBRARY}
+  ${ament_index_cpp_LIBRARIES}
+  ${rcutils_LIBRARIES}
+  ${TinyXML2_LIBRARY}
+  ${class_loader_LIBRARY}
+  ${console_bridge_LIBRARY}
+)
+
+foreach(_library ${_qt_gui_cpp_sip_LIBRARIES})
+if (EXISTS ${_library})
+list(APPEND qt_gui_cpp_sip_LIBRARIES ${_library})
+else()
+message(STATUS "Library with path \"${_library}\" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored")
```

Comment:
For which libraries is that actually the case? Either a library is needed or not - just skipping it when not available seems bogus.

Reply:
This message doesn't print out on the mac, but on Windows it filters out several libraries that are invalid library names. I'm not sure where it comes from. I agree its bogus, but this filtering needs to happen otherwise there are errors about not finding libraries.

Reply:
Which library names are failing? I would rather see the actual problem fixed then being worked around like this.

Reply:
I take it back, it does print out on MacOS

```
-- Unfiltered _qt_gui_cpp_sip_LIBRARIES /Users/brawner/workspace/ros2_core/install/ament_index_cpp/lib/libament_index_cpp.dylib;Poco::Foundation;console_bridge;/Users/brawner/workspace/ros2_core/install/class_loader/lib/libclass_loader.dylib;/Users/brawner/workspace/ros2_core/install/rcutils/lib/librcutils.dylib;/Users/brawner/workspace/ros2_core/install/rcutils/lib/librcutils.dylib;/Users/brawner/workspace/ros2_core/install/rcutils/lib/librcutils.dylib;tinyxml2;/usr/local/Frameworks/Python.framework/Versions/3.7/lib/libpython3.7m.dylib;/usr/local/lib/libtinyxml2.dylib;/Users/brawner/workspace/ros2_core/install/class_loader/lib/libclass_loader.dylib;/Users/brawner/workspace/ros2_core/install/console_bridge_vendor/lib/libconsole_bridge.dylib
-- Library with path "Poco::Foundation" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "console_bridge" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "tinyxml2" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Filtered _qt_gui_cpp_sip_LIBRARIES /Users/brawner/workspace/ros2_core/install/ament_index_cpp/lib/libament_index_cpp.dylib;Poco::Foundation;console_bridge;/Users/brawner/workspace/ros2_core/install/class_loader/lib/libclass_loader.dylib;/Users/brawner/workspace/ros2_core/install/rcutils/lib/librcutils.dylib;/Users/brawner/workspace/ros2_core/install/rcutils/lib/librcutils.dylib;/Users/brawner/workspace/ros2_core/install/rcutils/lib/librcutils.dylib;tinyxml2;/usr/local/Frameworks/Python.framework/Versions/3.7/lib/libpython3.7m.dylib;/usr/local/lib/libtinyxml2.dylib;/Users/brawner/workspace/ros2_core/install/class_loader/lib/libclass_loader.dylib;/Users/brawner/workspace/ros2_core/install/console_bridge_vendor/lib/libconsole_bridge.dylib

```

Reply:
On Ubuntu it can't find Poco::Foundation, console_bridge, stdc++fs

Reply:
I think it's because these are libraries that are created in cmake with the EXPORT flag, and thus useable in something like ```target_link_libraries```, but that's not sufficient for sip building.

Reply:
On Windows:

```
-- Library with path "Poco::Foundation" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "console_bridge" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "debug" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "optimized" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "debug" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
-- Library with path "optimized" was supposed to be linked, but the path is poorly formed or it does not exist. Therefore it won't be found by sip so it will be ignored
```

The debug/optimized library names are included from  TinyXML2

```
TINYXML2_LIBRARIES=debug;C:/ProgramData/Chocolatey/lib/tinyxml2/lib/tinyxml2d.lib;optimized;C:/ProgramData/Chocolatey/lib/tinyxml2/lib/tinyxml2.lib
```